dbms/src/Storages/S3/S3Common.cpp

```diff
@@ -627,11 +627,13 @@ void listPrefix(
 {
     Stopwatch sw;
     Aws::S3::Model::ListObjectsV2Request req;
-    req.WithBucket(client.bucket()).WithPrefix(client.root() + prefix);
+    bool is_root_single_slash = client.root() == "/";
+    // If the `root == '/'`, don't prepend the root to the prefix, otherwise S3 list doesn't work.
+    req.WithBucket(client.bucket()).WithPrefix(is_root_single_slash ? prefix : client.root() + prefix);
```

Comment:
maybe also add a check in the following codes that `root` can not start with **multiple** `/`?

https://github.com/pingcap/tiflash/blob/91454f6a75039a8fda0629153cd6752572d6f44b/dbms/src/Storages/S3/S3Common.cpp#L269